ocs_ci/helpers/dr_helpers.py

```diff
@@ -1082,3 +1095,213 @@ def verify_drpolicy_cli(switch_ctx=None):
+    multicluster_engine_yaml = tempfile.NamedTemporaryFile(
+        mode="w+", prefix="multiengine", delete=False
+    )
+    yaml_serialized = yaml.dump(resource)
+    multicluster_engine_yaml.write(yaml_serialized)
+    multicluster_engine_yaml.flush()
+    run_cmd(f"oc apply -f {multicluster_engine_yaml.name}")
+    multicluster_engine.wait_for_phase("Available")
+    config.switch_ctx(old_ctx)
+
+
+def create_dpa(bucket_name):
+    """
+    create DPA
+    OADP will be already installed when we enable backup flag
+    Here we will create dataprotection application and
+    update bucket name and s3 storage link
+    Args:
+        bucket_name (str): Name of the Bucket
+    """
+    oadp_data = templating.load_yaml(constants.ACM_DPA)
+    oadp_data["spec"]["backupLocations"][0]["velero"]["objectStorage"][
+        "bucket"
+    ] = bucket_name
+    oadp_yaml = tempfile.NamedTemporaryFile(mode="w+", prefix="oadp", delete=False)
+    templating.dump_data_to_temp_yaml(oadp_data, oadp_yaml.name)
+    run_cmd(f"oc create -f {oadp_yaml.name}")
+    # Validation
+    validate_dpa()
+
+
+@retry((CommandFailed, ACMClusterConfigurationException), tries=10, delay=10)
+def validate_dpa():
+    """
+    Validate
+    1. 3 restic pods
+    2. 1 velero pod
+    3. backupstoragelocation resource in "Available" phase
+    """
+    # Check restic pods.
+    # Restic pods have been renamed to node-agent after oadp 1.2
+    oadp_version = get_oadp_version()
+
+    if version.compare_versions(f"{oadp_version} >= 1.2"):
+        restic_pod_prefix = "node-agent"
+    else:
+        restic_pod_prefix = "restic"
+    restic_list = get_pods_having_label(
+        f"name={restic_pod_prefix}", constants.ACM_HUB_BACKUP_NAMESPACE
+    )
+    if len(restic_list) != constants.RESTIC_POD_COUNT:
+        raise ACMClusterConfigurationException("restic pod count mismatch")
+    for pod in restic_list:
+        if pod["status"]["phase"] != "Running":
+            raise ACMClusterConfigurationException("restic pod not in 'Running' phase")
+
+    # Check velero pod
+    veleropod = get_pods_having_label(
+        "app.kubernetes.io/name=velero", constants.ACM_HUB_BACKUP_NAMESPACE
+    )
+    if len(veleropod) != constants.VELERO_POD_COUNT:
+        raise ACMClusterConfigurationException("Velero pod count mismatch")
+    if veleropod[0]["status"]["phase"] != "Running":
+        raise ACMClusterConfigurationException("Velero pod not in 'Running' phase")
+
+    # Check backupstoragelocation resource in "Available" phase
+    backupstorage = ocp.OCP(
+        kind="BackupStorageLocation",
+        resource_name="default",
+        namespace=constants.ACM_HUB_BACKUP_NAMESPACE,
+    )
+    resource = backupstorage.get()
+    if resource["status"].get("phase") != "Available":
+        raise ACMClusterConfigurationException(
+            "Backupstoragelocation resource is no in 'Avaialble' phase"
+        )
+    logger.info("Dataprotection application successful")
```

Comment:
Post this, we should also add a check for policy compliance on both ACMs and proceed with hub recovery if it's complaint.

```
amagrawa:~$ oc get policy backup-restore-enabled -n open-cluster-management-backup
NAME                     REMEDIATION ACTION   COMPLIANCE STATE   AGE
backup-restore-enabled   inform               Compliant          10m
```

Reply:
This check can be performed post backupschedule creation and not when DPA validation i guess, Please correct me if i am wrong